genie-web/src/main/java/com/netflix/genie/web/tasks/job/JobCompletionService.java

```diff
@@ -154,6 +156,29 @@ void handleJobCompletion(final JobFinishedEvent event) throws GenieException {
         final Map<String, String> tags = MetricsUtils.newSuccessTagsMap();
 
         try {
+            /*
+             * Get the cluster and command chosen for the job for tagging metrics.
+             * Swallowing any exception in getting those as this step is not
+             * critical and should not show that the entire job failed
+             */
+            Cluster cluster;
```

Comment:
Strange, you don't get a warning that these can be final?

Reply:
Java complains if I declare as final that the variable might not be initialized